src/utils/map-state.js

```diff
@@ -0,0 +1,332 @@
+import PropTypes from 'prop-types';
+import {PerspectiveMercatorViewport} from 'viewport-mercator-project';
+import assert from 'assert';
+
+// MAPBOX LIMITS
+const MAX_PITCH = 60;
+const MAX_ZOOM = 20;
+
+function mod(value, divisor) {
+  const modulus = value % divisor;
+  return modulus < 0 ? divisor + modulus : modulus;
+}
+
+const propTypes = {
```

Comment:
Something about putting the prop types here doesn't feel quite right.

The MapState class is React independent, and PropTypes is still a React-related thing (it is certainly intended for React here).

Maybe just separate in two utility files map-state and map-props?